### ISSUE TITLE
NativeAOT: Type is not pre-initialized

### ISSUE THREAD

**Issue by EgorBo** [dotnet/runtime#83043]
```csharp
internal class Program
{
    struct MyStruct
    {
        public string Name;
    }

    static readonly MyStruct MyStr = new() { Name = "Hey!" };

    [MethodImpl(MethodImplOptions.NoInlining)]
    static MyStruct Test() => MyStr;
}
```
Codegen for `Test()`:
```asm
; Method Program:Test():Program+MyStruct
G_M41189_IG01:              ;; offset=0000H
       4883EC28             sub      rsp, 40
G_M41189_IG02:              ;; offset=0004H
       E800000000           call     CORINFO_HELP_READYTORUN_GCSTATIC_BASE
       488B4010             mov      rax, gword ptr [rax+10H]
G_M41189_IG03:              ;; offset=000DH
       4883C428             add      rsp, 40
       C3                   ret      
; Total bytes of code: 18
```
It seems that `_compilation.HasLazyStaticConstructor(field.OwningType)` returns true for `Program`

cc @MichalStrehovsky

**Comment by MichalStrehovsky** [comment]
Value types are represented as byte arrays within the interpreter. It's hard to add object references to it.

The features the interpreter supports follow the pareto principle.

**Comment by jkotas** [comment]
Similar issue for initialization of reference types: https://github.com/dotnet/runtime/issues/82993

**Comment by EgorBo** [comment]
@am11 yep, this issue is exactly what prevents NativeAOT to generate optimimal codegen for `RuntimeInformation.IsOSPlatform` API. I guess it's not super important because it's a sort of a legacy API (although, still tons of usages).

**Comment by am11** [comment]
@EgorBo, is it just this API, or is it part of emerging value-type pattern relying on FOH in static contexts?

**Comment by jkotas** [comment]
> or is it part of emerging value-type pattern relying on frozen strings in static contexts?

It is an example of code where tiered JITing has an advantage. Code that takes advantages of unique tiered JITing capabilities does not always work in the most optimal way with AOT, and vice versa.

In theory, this example can be fixed in the AOT interpreter. Most readonly statics instances that people expect to be optimized by tiered JIT are not like that.
